server/jamovi/server/dataset/duckdataset.py

```diff
@@ -555,3 +596,20 @@ def set_weights(self, weights_id: int) -> None:
     def refresh_filter_state(self) -> None:
         # TODO
         pass
+
+    def as_sql_string(self, value, data_type,measure_type):
+        if self.is_null_value(value, data_type):
+            return NULL_VALUES[data_type][measure_type]
+        # TODO: Add more cases for other data types
+        ...
+
+        return value
+
+    def is_null_value(self, value, data_type):
+        if data_type is DataType.DECIMAL or data_type is DataType.INTEGER:
+            import math # Note: Obviously not a great place for the import, but okay for now
+            return math.isnan(value)
```

Comment:
for INTEGER, a missing value is denoted with that `-21....` number ... `INT_MIN` for 32 bit ints.